parent: !include default.yaml
robot_interface:
  base:
    plugin: CmdVelInterface
    cmd_vel_topic: /pr2/cmd_vel
    drive:
      type: OmniDrive
      parent_link_name: odom_combined
      child_link_name: base_footprint
  body:
    plugin: SendFollowJointTrajectory
    namespace: /pr2/whole_body_controller/follow_joint_trajectory
    state_topic: /pr2/whole_body_controller/state
    fill_velocity_values: False
plugins:
  SyncTfFrames:
    frames:
      - [map, odom_combined]
general_options:
  joint_limits: # giskard will use the min of this number and limits from the urdf
    velocity:
      linear: # m/s
        default: 1
        override: {}
      angular: # rad/s
        default: 1
        override:
          head_pan_joint: 0.5
          head_tilt_joint: 0.5
    acceleration:
      linear: # m/s**2
        default: 1e3
        override: { }
      angular: # rad/s**2
        default: 1e3
        override: { }
    jerk:
      linear: # m/s**3
        default: 30
        override:
          torso_lift_joint: 5
      angular: # rad/s**3
        default: 30
        override: {}
#    snap:
#      linear: # m/s**4
#        default: 250
#        override: { }
#      angular: # rad/s**3
#        default: 250
#        override: { }
plugins:
  LoopDetector:
    precision: 4 # decimal places
collision_avoidance:
  external_collision_avoidance:
    default:
      hard_threshold: 0.0
      soft_threshold: 0.1
    override:
      4eef: &4eef
        number_of_repeller: 4
        hard_threshold: 0.00
        soft_threshold: 0.05
      2eef: &2eef
        number_of_repeller: 2
        hard_threshold: 0.00
        soft_threshold: 0.05
      50mm: &50mm
        hard_threshold: 0.00
        soft_threshold: 0.05
      25mm: &25mm
        number_of_repeller: 2
        hard_threshold: 0.00
        soft_threshold: 0.025
      r_wrist_roll_joint: *4eef
      l_wrist_roll_joint: *4eef
      r_forearm_roll_joint: *25mm
      l_forearm_roll_joint: *25mm
      r_elbow_flex_joint: *50mm
      l_elbow_flex_joint: *50mm
      r_wrist_flex_joint: *2eef
      l_wrist_flex_joint: *2eef
  self_collision_avoidance:
    ignore: # list pairs of links that should be removed from the self collision avoidance
      - bl_caster_l_wheel_link
      - bl_caster_r_wheel_link
      - bl_caster_rotation_link
      - br_caster_l_wheel_link
      - br_caster_r_wheel_link
      - br_caster_rotation_link
      - fl_caster_l_wheel_link
      - fl_caster_r_wheel_link
      - fl_caster_rotation_link
      - fr_caster_l_wheel_link
      - fr_caster_r_wheel_link
      - fr_caster_rotation_link
      - l_shoulder_lift_link
      - r_shoulder_lift_link
      - base_link
    add: [ ] # list pairs of links that should be added to the self collision matrix
    number_of_repeller: 1 # each movable joint gets pushed away from the X objects closest to it
